src/DeviceInput/InputDevices/deviceSourceManager.ts

```diff
@@ -133,6 +147,24 @@ export class DeviceSourceManager implements IDisposable {
         return this._devices[deviceType].filter((source) => { return !!source; });
     }
 
+    /**
+     * Returns a read-only list of all available devices
+     * @returns Read-only array with active devices
+     */
+    public getListofDevices(): ReadonlyArray<{ deviceType: DeviceType, deviceSlot: number }> {
+        let deviceArray: Array<{ deviceType: DeviceType, deviceSlot: number }> = new Array<{ deviceType: DeviceType, deviceSlot: number }>();
+        this._devices.forEach((deviceSet) => {
+            deviceSet.forEach((device) => {
+                const deviceType: DeviceType = device.deviceType;
+                const deviceSlot: number = device.deviceSlot;
+
+                deviceArray.push({ deviceType, deviceSlot });

```

Comment:
Should we return a copy or should we keep it internally as it is readonly ?

Reply:
With the switch to DeviceSource, I'll just be returning a list of refs to existing DeviceSources